src/services/tokens.ts

```diff
@@ -15,6 +18,55 @@ export interface Token {
   jti: string;
 }
 
+function generateRefreshToken(user: User): string {
+  const username = user.Username;
+  const refreshToken = jwt.sign({ user: { username } }, REFRESH_TOKEN_SECRET);
+  refreshTokens[username] = refreshToken;
+  return refreshToken;
+}
+
+function generateIdToken(
+  user: User,
+  authTime: number,
+  userPoolId: string,
+  clientId: string,
+  eventId: string
+) {
+  const attributes: any = {};
+  user.Attributes.forEach((attribute) => {
+    const { Value } = attribute;
+    let { Name } = attribute;
+    if (
+      Name.toLowerCase().startsWith("custom") &&
+      Name.split(":").length === 2
+    ) {
+      Name = Name.split(":")[1];
+    }
+    attributes[Name] = Value;
+  });
```

Comment:
Attributes we provide on the IdToken were not populated. This collects the attributes to add them to the IdToken